pkg/identity/certificated.go

```diff
@@ -102,17 +102,26 @@ func Certificated(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (er
 					if err := w.AddBytes(outPath, 0644, roleCertPEM); err != nil {
 						return errors.Wrap(err, "unable to save x509 role cert")
 					}
+
+					// always output role cert key file to prevent unexpected key rotation when using external key
+					if id == nil || idConfig.RoleCertKeyFileOutput {
```

Comment:
```suggestion
					if id != nil || idConfig.RoleCertKeyFileOutput {
```
Might be wrong but isn't it `id == nil`